plugins/Login/Security/BruteForceDetection.php

```diff
@@ -0,0 +1,134 @@
+<?php
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+namespace Piwik\Plugins\Login\Security;
+
+use Piwik\Common;
+use Piwik\Date;
+use Piwik\Db;
+use Piwik\Plugins\Login\SystemSettings;
+
+class BruteForceDetection {
+
+    private $minutesTimeRange;
+    private $maxLogAttempts;
+
+    private $table = 'brute_force_log';
+    private $tablePrefixed = '';
+
+    /**
+     * @var SystemSettings
+     */
+    private $settings;
+
+    public function __construct(SystemSettings $systemSettings)
+    {
+        $this->tablePrefixed = Common::prefixTable($this->table);
+        $this->settings = $systemSettings;
+        $this->minutesTimeRange = $systemSettings->loginAttemptsTimeRange->getValue();
+        $this->maxLogAttempts = $systemSettings->maxFailedLoginsPerMinutes->getValue();
+    }
+
+    public function isEnabled()
+    {
+        return $this->settings->enableBruteForceDetection->getValue();
+    }
+
+    public function addFailedLoginAttempt($ipAddress)
```

Comment:
I'm wondering, could this be a potential ddos attack vector? If someone purposefully sends tons of tracker requests w/o a token auth, would it fill up the table w/ meaningless data? I suppose they would have to vary their IP addresses, or use multiple ip addresses and time things right to add the junk data. Not sure exactly what kind of effect this would have.

Reply:
Depends... if you have access to someones WIFI, you could possibly try lots of failed logins to log out the organisation...Or depending how Matomo is configured, you could maybe set a custom IP in the header if reads the IP from a different header.

Otherwise re meaningless data... you could also send heaps of meaningless data to the tracker API etc. I don't think that's an issue?

Reply:
Was wondering if it would affect the tracking latency, but sounds like it won't